mk/config.mk

```diff
@@ -648,6 +648,10 @@ endif
 # Enables backwards compatible derivation of RPMB and SSK keys
 CFG_CORE_HUK_SUBKEY_COMPAT ?= y
 
+# Use SoC specific tee_otp_get_die_id implementation for SSK key generation.
```

Comment:
- `tee_otp_get_die_id implementation()`
- You mention only the SSK but the above comment mentions both RPMB key and SSK.
- Without going through the history of the changes in the code, it is hard to understand when one would want to enable this config. How about explaining it in more details in the commit description at least? Similar to what you did in https://github.com/OP-TEE/optee_os/pull/4989#issuecomment-975439147.

Reply:
* Done
* RPMB and SSK have different generation schemes.
  Old scheme:
  `SSK = HMAC_SHA256(HUK, Chip_ID || "ONLY_FOR_tee_fs_ssk")`
  `RPMB_KEY = HMAC_SHA256(HUK, eMMC_CID)`
  
  New scheme:
  `SSK = HMAC_SHA256(HUK, HUK_SUBKEY_SSK)`
  `RPMB_KEY = HMAC_SHA256(HUK, HUK_SUBKEY_RPMB || eMMC_CID)`
  
  Where:
  `HUK` - hardware unique key
  `Chip_ID` - `get_otp_die_id`
  `HMAC_SHA256(key, value)`
  
  `Chip_ID` was only used in the old implementation of the SSK generation scheme.
* Done.

P.S. Could you please explain what was the reason for changing the generation scheme? Because I do not see a big difference but I might have missed something. Thank you in advance.

Reply:
Thanks for claryfing.

> P.S. Could you please explain what was the reason for changing the generation scheme? Because I do not see a big difference but I might have missed something. Thank you in advance.

I believe the main reason is to avoid code that would manipulate the HUK in an unsafe manner and could potentially leak information about it. By providing the `huk_subkey_derive()` interface, this risk is minimized. See the description of commits fa0525fa and df91a522. @jenswi-linaro

Reply:
That and also to have a structured way of generating a suitable hardware unique key for each situation to avoid key reuse and such.